test-utils/src/async_bag.rs

```diff
@@ -1,9 +1,25 @@
 use std::{ops::Deref, sync::Arc};
 use tokio::sync::{Mutex, MutexGuard};
 
-#[derive(Clone, Debug)]
+/// Wrapper for `Arc<Mutex<T>>` used to make testing easier.
+#[derive(Debug)]
 pub struct AsyncBag<T>(Arc<Mutex<T>>);
```

Comment:
whats the point of this?

Reply:
side note: The tokio mutex is an antipattern. Its extremely rare that it should be used.